### ISSUE TITLE
SC + Yokozuna?

### ISSUE THREAD

**Issue by rzezeski** [basho/yokozuna#353]
Yesterday @jonmeredith @jtuple and others discussed interaction between Strong Consistency (SC) and Yokozuna. No matter what there will be no strongly consistent indexes in 2.0. The indexes will be eventually consistent/async to the KV writes. For the most part we are confident things will mostly work it sounds like there are a few edge cases we need to test as well as some extra code around delete operations (apparently SC uses a sentinel object value for delete ATM). This is a marker to make sure we do that work for 2.0 (or document that SC + YZ is untested in 2.0 and therefore undefined behavior).

**Comment by rzezeski** [comment]
There has been no time to make any progress on this issue.  Should it still be 2.0-RC?

@jonmeredith @jtuple @coderoshi

**Comment by jonmeredith** [comment]
It should either work, or not work with an error message when enabling search and SC on a bucket. Given how low on time we are, I pick whichever is lowest effort.

**Comment by jtuple** [comment]
This is on my list to test after the remaining SC issues get resolved this week. As Jon mentioned, **something** needs to happen since at the very least we need to disable indexing of consistent bucket types if things don't work. In any case, first step is writing a test and seeing what happens -- which I'll hopefully do next week.

**Comment by coderoshi** [comment]
Has any progress been made on this, or do we need to start considering @jonmeredith's error message backup plan?

@jtuple @rzezeski

**Comment by rzezeski** [comment]
@coderoshi I've done nothing with this.

**Comment by coderoshi** [comment]
I tested YZ against the set of ensemble test. They required some modifications (namely, objects needed content types, key values had to be valid UTF8, yz service activation, waits, and indexing), but passed. The known failure case was deletes, which were resolved in #404 and tested basho/riak_test#650. There are a couple edge cases which should be documented, basho/basho_docs#1151, but that shouldn't hold up RC1
